Document Engine APIs (#1003)
hyperledger/besu-docs: docs/Reference/Engine-API-Methods.md

---
description: Engine API methods reference
---

# Engine API methods

> The Engine API allows communication between [consensus and execution layers](../Concepts/Merge.md#execution-and-consensus-clients) of the [post-Merge](../Concepts/Merge.md) Ethereum Client.

Reviewer: ```suggestion
After [The Merge](../Concepts/Merge.md), consensus and execution clients communicate with each other using the Engine API.
When running Besu as an execution client, [use these API calls](../HowTo/Interact/APIs/Engine-API.md) to communicate with a consensus client.
```

Always link back to related pages.

Author: Resolve

> See [Ethereum Engine API specification](https://github.com/ethereum/execution-apis/blob/main/src/engine/specification.md) for more information.

Reviewer: ```suggestion
See the [Ethereum Engine API specification](https://github.com/ethereum/execution-apis/blob/main/src/engine/specification.md) for more information.
```

Author: Fixed

## Methods

### `engine_exchangeTransitionConfigurationV1`

> Sends to the consensus client to verify the configuration between both clients.

Reviewer: Sends the execution client configuration?

Author: Fixed

!!! note

>     The Engine runs this call every 60 seconds in the background.

Reviewer: Besu / the execution client runs?

Author: Fixed

              "receiptsRoot": "0x56e81f171bcc55a6ff8345e692c0f86e5b48e01b996cadc001622fb5e363b421",
              "logsBloom": "0x00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000",
              "prevRandao": "0x0000000000000000000000000000000000000000000000000000000000000000",
              "blockNumber": "0x1",
              "gasLimit": "0x1c9c380",
              "gasUsed": "0x0",
              "timestamp": "0x5",
              "extraData": "0x",
              "baseFeePerGas": "0x7",
              "blockHash": "0x3559e851470f6e7bbed1db474980683e8c315bfce99b2a6ef47c057c04de7858",
              "transactions": []
            }
        }
        ```

### `engine_newPayloadV1`

> Executes the payload with the consensus client

Reviewer: ```suggestion
Executes the payload with the consensus client.
```

Author: Fixed
